src/app/static/src/app/components/WarningAlert.vue

```diff
@@ -0,0 +1,175 @@
+            <div :style="containerBox">
+                <div ref="warningBox" id="warningBox">
+                    <div v-for="(value, key) in filteredWarnings" :key="key">
+                        <h4 class="alert-heading pt-2">
+                            <alert-triangle-icon size="1.5x" class="custom-class mr-1 mb-1"></alert-triangle-icon>
+                            {{ headerText(key) }}
+                        </h4>
+                        <ul class="mb-0">
+                            <li v-for="warning in value" :key="warning.text">{{ warning.text }}</li>
+                        </ul>
+                    </div>
+                </div>
+                <div ref="incHeader" class="invisible">
+                    <h4 class="alert-heading pt-2">
+                        <alert-triangle-icon size="1.5x" class="custom-class mr-1 mb-1"></alert-triangle-icon>
+                        Hidden header
+                    </h4>
+                    <p ref="line" class="mb-0">...</p>
+                </div>
+            </div>
+            <div v-if="warningsLengthy" id="showToggle">
+                <p v-if="!showFullBox" class="ml-4 mb-0">...</p>
+                <button @click="toggleShowFullBox" class="btn btn-link alert-link">{{ buttonText }}</button>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue";
+    import { AlertTriangleIcon } from "vue-feather-icons";
+    import i18next from "i18next";
+    import { mapStateProp } from "../utils";
+    import { RootState } from "../root";
+    import { Language } from "../store/translations/locales";
+import { switches } from "../featureSwitches";
+
+    interface Props {
+        warnings: Warnings;
+        maxLines: number;
+    }
+
+    interface Data {
+        showFullBox: boolean;
+        fullBoxHeight: number;
+        lineHeight: number;
+        headerHeight: number;
+    }
+
+    interface Methods {
+        toggleShowFullBox: () => void;
+        updateDimensions: () => void;
+        headerText: (key: string) => string;
+    }
+
+    interface Computed  {
+        currentLanguage: Language;
+        renderedBoxHeight: number;
+        maxBoxHeight: number;
+        containerBox: {
+            height: string,
+            overflowY: string
+        };
+        filteredWarnings: Warnings;
+        warningsLengthy: boolean;
+        showAlert: boolean;
+        buttonText: string;
+    }
+
+    interface Warning {
+        text: string;
+        locations: ("model_options" | "model_fit" | "model_calibrate" | "review_output" | "download_results")[];
+    }
+
+    interface Warnings {
+        [key: string]: Warning[];
+    }
+
+    export default Vue.extend<Data, Methods, Computed, Props>({
+        name: "WarningAlert",
+        props: {
+            warnings: Object,
+            maxLines: {
+                default: 3,
+                required: false,
+                type: Number
+            }
```

Comment:
The user has the option to configure the number of lines (after the first warning header) that will be displayed. If the height of the warnings are more than this number, the last line will be replaced with an ellipsis and the show more button will appear.